File: go/965.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isUnivalTree(root *TreeNode) bool {
	if root == nil {
		return true
	} else if root.Left == nil && root.Right == nil {
		return true
	} else if root.Left == nil {
		return root.Val == root.Right.Val && isUnivalTree(root.Right)
	} else if root.Right == nil {
		return root.Val == root.Left.Val && isUnivalTree(root.Left)
	} else {
		return root.Val == root.Right.Val && isUnivalTree(root.Right) && root.Val == root.Left.Val && isUnivalTree(root.Left)
	}
}

File: java/965.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isUnivalTree(TreeNode root) {
        if(root == null){
            return true;
        }
        if(root.left == null && root.right == null){
            return true;
        }
        if(root.left == null){
            return root.val == root.right.val && isUnivalTree(root.right); 
        }
        if(root.right == null){
            return root.val == root.left.val && isUnivalTree(root.left);
        }
        return root.val == root.right.val && isUnivalTree(root.right) && root.val == root.left.val && isUnivalTree(root.left);
    }
}